back/src/main/java/ro/utcn/sd/he/assignment1/controller/AnswerController.java

```diff
@@ -0,0 +1,100 @@
+import ro.utcn.sd.he.assignment1.command.AddAnswerCommand;
+import ro.utcn.sd.he.assignment1.command.Command;
+import ro.utcn.sd.he.assignment1.command.EditAnswerCommand;
+import ro.utcn.sd.he.assignment1.dto.AnswerVoteDTO;
+import ro.utcn.sd.he.assignment1.dto.VoteDTO;
+import ro.utcn.sd.he.assignment1.event.*;
+import ro.utcn.sd.he.assignment1.model.Answer;
+import ro.utcn.sd.he.assignment1.model.Question;
+import ro.utcn.sd.he.assignment1.model.User;
+import ro.utcn.sd.he.assignment1.service.AnswerService;
+import ro.utcn.sd.he.assignment1.service.QuestionService;
+import ro.utcn.sd.he.assignment1.service.UserService;
+import ro.utcn.sd.he.assignment1.service.VoteService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@RequiredArgsConstructor
+@RestController
+public class AnswerController {
+    private final AnswerService answerService;
+    private final UserService userService;
+    private final QuestionService questionService;
+    private final VoteService voteService;
+
+    private final SimpMessagingTemplate messagingTemplate;
+    private List<Command> commandList = new ArrayList<>();
+
+    @GetMapping("/answers")
+    public List<AnswerVoteDTO> getAnswersOf(@RequestParam int id) {
+        Question question = questionService.findById(id);
+        return answerService.findAnswersOf(question);
+    }
+
+    @PostMapping("/answer")
+    public AnswerVoteDTO saveAnswer(@RequestBody Answer answer) {
+        AddAnswerCommand command = answerService.saveAnswer(answer);
+        commandList.add(command);
+        AnswerVoteDTO output = AnswerVoteDTO.ofEntity(command.getAnswer(), voteService.getVoteCount(command.getAnswer()));
+
+        BaseEvent event = new AnswerCreatedEvent(output);
+        messagingTemplate.convertAndSend("/topic/events", event);
+        log.info("Got an event: {}.", event);
+
+        return output;
+    }
+
+    @PutMapping("/answer")
+    public AnswerVoteDTO editAnswer(@RequestBody Answer newAnswer) {
+        EditAnswerCommand command = answerService.editAnswer(newAnswer);
+        commandList.add(command);
+        AnswerVoteDTO output = AnswerVoteDTO.ofEntity(command.getNewAnswer(), voteService.getVoteCount(command.getNewAnswer()));
+
+        BaseEvent event = new AnswerEditedEvent(output);
+        messagingTemplate.convertAndSend("/topic/events", event);
+        log.info("Got an event: {}.", event);
+
+        return output;
+    }
+
+    @DeleteMapping("/answer")
+    public void deleteAnswer(@RequestParam int id) {
+        Command command = answerService.deleteAnswer(id);
+        commandList.add(command);
+
+        BaseEvent event = new AnswerDeletedEvent(id);
+        messagingTemplate.convertAndSend("/topic/events", event);
+
+    }
+
+    @PostMapping("/answers/vote")
+    public VoteDTO vote(@RequestParam(name = "answerId") int answerId, @RequestParam(name = "username") String username, @RequestParam(name = "vote") String vote) {
```

Comment:
In general, mixing query params and POST calls is not really used. Instead you should: take the user from the security context and take the answerId from a PathVariable (the URL should be `/answers/{answerId}/votes`).